apps/dashboard/src/components/workflow-row.tsx

```diff
@@ -9,47 +21,60 @@ import {
   DropdownMenuSeparator,
   DropdownMenuTrigger,
 } from '@/components/primitives/dropdown-menu';
-import { Link } from 'react-router-dom';
+import { HoverToCopy } from '@/components/primitives/hover-to-copy';
 import { TableCell, TableRow } from '@/components/primitives/table';
-import { useEnvironment } from '@/context/environment/hooks';
-import { WorkflowOriginEnum } from '@/utils/enums';
-import { buildRoute, LEGACY_ROUTES, ROUTES } from '@/utils/routes';
-import { Badge } from '@/components/primitives/badge';
-import { BadgeContent } from '@/components/primitives/badge';
-import { WorkflowStatus } from '@/components/workflow-status';
-import { WorkflowSteps } from '@/components/workflow-steps';
-import { WorkflowTags } from '@/components/workflow-tags';
-import { FaCode } from 'react-icons/fa6';
-import TruncatedText from '@/components/truncated-text';
 import {
   Tooltip,
   TooltipContent,
   TooltipPortal,
   TooltipProvider,
   TooltipTrigger,
 } from '@/components/primitives/tooltip';
-import { RiMore2Fill, RiPlayCircleLine } from 'react-icons/ri';
+import TruncatedText from '@/components/truncated-text';
+import { WorkflowStatus } from '@/components/workflow-status';
+import { WorkflowSteps } from '@/components/workflow-steps';
+import { WorkflowTags } from '@/components/workflow-tags';
+import { useEnvironment } from '@/context/environment/hooks';
+import { useDeleteWorkflow } from '@/hooks/use-delete-workflow';
 import { useSyncWorkflow } from '@/hooks/use-sync-workflow';
-import { HoverToCopy } from '@/components/primitives/hover-to-copy';
+import { WorkflowOriginEnum } from '@/utils/enums';
+import { buildRoute, LEGACY_ROUTES, ROUTES } from '@/utils/routes';
+import { ConfirmationModal } from './confirmation-modal';
 
 type WorkflowRowProps = {
   workflow: WorkflowListResponseDto;
 };
 
 export const WorkflowRow = ({ workflow }: WorkflowRowProps) => {
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
   const { currentEnvironment } = useEnvironment();
   const { safeSync, isSyncable, tooltipContent, PromoteConfirmModal } = useSyncWorkflow(workflow);
 
   const isV1Workflow = workflow.origin === WorkflowOriginEnum.NOVU_CLOUD_V1;
   const workflowLink = isV1Workflow
     ? buildRoute(LEGACY_ROUTES.EDIT_WORKFLOW, {
-        workflowSlug: workflow.slug,
+        workflowId: workflow._id,
       })
     : buildRoute(ROUTES.EDIT_WORKFLOW, {
         environmentSlug: currentEnvironment?.slug ?? '',
         workflowSlug: workflow.slug,
       });
 
+  const triggerWorkflowLink = isV1Workflow
+    ? buildRoute(LEGACY_ROUTES.TEST_WORKFLOW, { workflowId: workflow._id })
+    : buildRoute(ROUTES.TEST_WORKFLOW, {
+        environmentSlug: currentEnvironment?.slug ?? '',
+        workflowSlug: workflow.slug,
+      });
+
+  const { deleteWorkflow } = useDeleteWorkflow();
+
+  const onDeleteStep = async () => {
+    await deleteWorkflow({
+      workflowId: workflow._id,
+    });
+  };
```

Comment:
I'm missing here a toast for the successful deletion, but I don't see it in the designs either.

Reply:
Adding it, this be consistent with other places